drivers/accel/adxl355/iio_adxl355_trig.c

```diff
@@ -149,6 +195,45 @@ int iio_adxl355_trigger_init(struct adxl355_iio_trig **iio_trig,
 	return ret;
 }
 
+/***************************************************************************//**
+ * @brief Initialize ADXL355 software trigger.
+ *
+ * @param iio_trig   - The iio trigger structure.
+ * @param init_param - The structure that contains the sw trigger initial params.
+ *
+ * @return ret       - Result of the initialization procedure.
+*******************************************************************************/
+int iio_adxl355_software_trigger_init(struct adxl355_iio_trig **iio_trig,
+				      struct adxl355_iio_sw_trig_init_param *init_param)
+{
+	struct adxl355_iio_trig *trig_desc;
+	int ret;
+
+	trig_desc = (struct adxl355_iio_trig*)calloc(1, sizeof(*trig_desc));
+	if (!trig_desc)
+		return -ENOMEM;
+
+	if (!init_param->iio_desc) {
+		ret =-EINVAL;
+		goto error;
+	}
+	trig_desc->iio_desc = init_param->iio_desc;
+
+	if(!init_param->name) {
+		ret=-EINVAL;
+		goto error;
+	}
```

Comment:
These 2 checks can be made before `trig_desc` is allocated, which will simplify the code a bit. The same applies for the `iio_adxl355_trigger_init` as well.